Assets/MixedRealityToolkit.Tests/EditModeTests/InputSystem/InputSystemTests.cs

```diff
@@ -98,10 +89,76 @@ public void TestMixedRealityInputSystemDataProviders()
             Assert.IsEmpty(dataProviderAccess.GetDataProviders());
         }
 
+        [Test]
+        public void TestDataProviderRegisteration()
+        {
+            TestUtilities.InitializeMixedRealityToolkitAndCreateScenes();
+            MixedRealityToolkit.Instance.ActiveProfile.InputSystemProfile = AssetDatabase.LoadAssetAtPath<MixedRealityInputSystemProfile>(TestInputSystemProfilePath);
+
+            var inputSystem = new MixedRealityInputSystem(MixedRealityToolkit.Instance, MixedRealityToolkit.Instance.ActiveProfile.InputSystemProfile);
+            Assert.IsNotNull(inputSystem);
+            Assert.IsTrue(MixedRealityToolkit.Instance.RegisterService<IMixedRealityInputSystem>(inputSystem));
+
+            // Since EditMode, we have to auto-enable MRTK input system ourselves
+            MixedRealityToolkit.Instance.EnableAllServicesByType(typeof(IMixedRealityInputSystem));
+
+            Assert.AreEqual(1, MixedRealityServiceRegistry.GetAllServices().Count);
+            Assert.IsNotNull(MixedRealityToolkit.Instance.GetService<IMixedRealityInputSystem>());
+
+            var dataProviderAccess = (inputSystem as IMixedRealityDataProviderAccess);
+            Assert.IsNotNull(dataProviderAccess);
+
+            var dataProvider = dataProviderAccess.GetDataProvider<TestInputDataProvider>();
+            Assert.IsNotNull(dataProvider);
+            Assert.IsTrue(dataProvider.IsInitialized);
+            Assert.IsTrue(dataProvider.IsEnabled);
+
+            inputSystem.Disable();
+            Assert.IsFalse(dataProvider.IsEnabled);
+
+            inputSystem.Enable();
+            // We still have reference to old dataprovider, check still disabled
+            Assert.IsFalse(dataProvider.IsEnabled);
+
+            // dataProvider has been unregistered in Disable and new one created by Enable.
+            dataProvider = dataProviderAccess.GetDataProvider<TestInputDataProvider>();
+            Assert.IsNotNull(dataProvider);
+            Assert.IsTrue(dataProvider.IsEnabled);
+
+            inputSystem.Reset();
+            LogAssert.Expect(LogType.Log, "TestDataProvider Reset");
+            Assert.IsFalse(dataProvider.IsEnabled);
+
+            // dataProvider has been unregistered and newly created in Reset
+            dataProvider = dataProviderAccess.GetDataProvider<TestInputDataProvider>();
+            Assert.IsNotNull(dataProvider);
+            Assert.IsTrue(dataProvider.IsInitialized);
```

Comment:
>ssert.IsTrue(dataProvider.IsInitialized); [](start = 13, length = 41)

Would it also be worth asserting that dataProvider.IsEnabled (i.e. if it gets created enabled?)